File: prime.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    cout << "Prime Numbers between 20 and 30 are:\n";

    for (int i = 20; i <= 30; i++) //loop to check for each number in the range

    {
        int ctr = 0; //to maintain factor count

        for (int j = 2; j <= sqrt(i); j++) //checking for factors

        {
            if (i % j == 0)

                ctr = 1; //increasing factor count when found
        }

        if (ctr == 0) //checking and printing prime numbers

            cout << i << " ";
    }

    return 0;
}

// To check given number is prime or not?         
// #include<iostream>
// using namespace std;
// int main()
// {
//     int n,i,m=0,flag=0;
//     cout<<"Enter the number to check prime : ";
//     cin>>n;
//     m=n/2;
//     for(i=2;i<=m;i++)
//     {
//         if(n%i==0)
//         {
//             cout<<"Number is not Prime."<<endl;
//             flag=1;
//             break;
//         }
//     }
//     if(flag==0){
//         cout<<"Number is Prime."<<endl;
//     return 0;
//     }
// }

File: Prime.java
import java.lang.Math;
public class arrays {
    public static String isPrime(int a){
        int i, m = 0, flag = 0;
        m = a / 2;
        if (a == 0 || a == 1) {
            return "false";
        } else {
            for (i = 2; i <= m; i++) {
                if (a % i == 0) {
                    flag = 1;
                    return "false";
                    // break;
                }
            }
            if (flag == 0) {
                return "true";
            }
        }
        return null;
    }
    public static void main(String[] args) {
        int a = 4;
        int b = 16;
        
        if(isPrime(a)=="true" && isPrime(b)=="true"){
            System.out.println(Math.min(a, b));
        }else if(isPrime(a)=="true" || isPrime(b)=="true"){
            System.out.println(Math.max(a, b));
        }else{
            System.out.println("Both are not Prime Numbers.");
        }
    }
}
